snapcraft/internal/errors.py

```diff
@@ -243,7 +269,10 @@ def __init__(self, error=None):
 class MissingGadgetError(SnapcraftError):
 
     fmt = (
-        'When creating gadget snaps you are required to provide a gadget.yaml file\n'  # noqa
+        'Failed to generate snap metadata: '
```

Comment:
We actually have three exceptions that all have the same "what." This suggests a common base class to me: `MetadataGenerationError` or some such thing.